### ISSUE TITLE
Column start greater than Error

### ISSUE THREAD

**Issue by baldwmic** [AtomLinter/linter-eslint#689]
# Error
## `Column start greater than line length`
![eslint errors_censored](https://cloud.githubusercontent.com/assets/10538297/18392956/bb4763d2-7681-11e6-867e-ed707f91ea80.jpg)

## Similar Issues
I've searched for similar issues [here](https://github.com/AtomLinter/linter-eslint/issues?utf8=%E2%9C%93&q=column%20start%20greater%20than) and [here](https://github.com/AtomLinter/linter-stylelint/issues?utf8=%E2%9C%93&q=column%20start%20greater%20than) but can't seem to figure this out.

# Reproduce
- files: [file.js, .eslintrc, package.json](https://drive.google.com/folderview?id=0B8rIACP33UnHWkh2S2c4M18zV0U&usp=sharing)

**Comment by baldwmic** [comment]
I ran the following command: 
```
./node_modules/.bin/eslint src *.js --quiet
``` 
and got the following output:

> The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.

Might be something in my `.eslintrc` file as related to [this issue](https://github.com/airbnb/javascript/issues/978)

**Comment by Arcanemagus** [comment]
Looking into this, btw for future reference: You can attach zip files directly in GitHub now 😉.

**Comment by Arcanemagus** [comment]
So interestingly you _should_ be getting a `Error running ESLint Error: EPERM: operation not permitted, write` due to that message you posted above. https://github.com/atom/atom/issues/12504 is where working around that limitation is being tracked.

You can work around that by adding `"react/require-extension": 0` to your `"rules"` section of your `.eslintrc`.

**Comment by Arcanemagus** [comment]
After turning that rule off so it will even run, I'm not able to reproduce the error you are seeing with the files you provided 😕.

Can you try turning that rule off as described above and then trying to reproduce this locally again?

**Comment by Arcanemagus** [comment]
@vadim since you 👍 this issue, can you file a _new_ issue, with the same set of files @baldwmic provided (eg. `foo.js`, `package.json`, `.eslintrc.js`)? It's highly unlikely you are hitting the _exact_ same issue.

**Comment by baldwmic** [comment]
@Arcanemagus i've tried adding `"react-require-extension": 0` to my `"rules"` in `.eslintrc` and i'm still encountering the same error in the same javascript file 😢 

Strangely enough, I don't see a `Error running ESLint Error: EPERM: operation not permitted, write`

**Comment by Arcanemagus** [comment]
Hmmm, are you seeing it on the file in your project, or in an isolated folder containing _just_ the 3 files you linked?

If you add `"root": true` to the `.eslintrc` file does it change anything? It's possible the rule causing the error is coming from a parent `.eslintrc` file.

**Comment by IanVS** [comment]
@baldwmic are you getting any linting errors?  Meaning, not the big red boxes, but an actual eslint rule error?

**Comment by digitalcatnip** [comment]
@IanVS I'm experiencing this issue constantly while writing code - it's generating eslint errors because the code isn't fully completed when I stop typing.  The most frequent error is "missing semicolon" because as I said, I haven't finished the line yet.  The other error would be missing import because I haven't actually imported the file for the new code yet.

This just started happening about 2 days ago, and it's incredibly annoying so let me know what you need to help repro.

<img width="1039" alt="screen shot 2016-09-09 at 1 59 10 pm" src="https://cloud.githubusercontent.com/assets/14783203/18397332/b0a47474-7695-11e6-9cb3-000d3a46691d.png">

**Comment by IanVS** [comment]
What is the error you are getting two lines above the semi?  `HLOChatMessage()`

**Comment by Arcanemagus** [comment]
@digitalcatnip When you see that message can you _immediately_ save the file, then save it again to see if it triggers the error again? If so can you file a _new_ issue, as generally each of these invalid range errors are unique.

**Comment by IanVS** [comment]
@digitalcatnip in your new issue, can you provide the file which is causing the error?  As small of a file as possible, if you could.

**Comment by baldwmic** [comment]
@Arcanemagus @IanVS 

# New Steps
I've taken a few steps to help suss out whether this is an issue specific to my project or to `atom` and/or `eslint`.

## Isolated Dependencies
1. Added the `"rules": { "react/require-extension": 0, }` [workaround](https://github.com/atom/atom/issues/12504) to `.eslintrc` file
2. Modified the JavaScript file to have zero imports from other files.
3. Added the files as a separate, isolated project folder in Atom.

# Error
While editing the single JavaScript file in Atom, here is what I encounter:

![atom_linter](https://cloud.githubusercontent.com/assets/10538297/18397867/1cd577ae-7698-11e6-96c4-895aa177ffd3.gif)

## Notes
- `.eslintrc` file has `"rules": { "react/require-extension": 0, }` workaround
- there are zero `eslint` rule errors when I begin editing the file
- the `Column start greater than line length` error still appears

# Reproduce
1. Download [the zipped files](https://github.com/AtomLinter/linter-eslint/files/464551/project.zip)
I'm using
2. `cd ~/Downloads && unzip project.zip`
3. `npm install`
4. `Add Project Folder` of the unzipped files using Atom
5. Edit the `file.js` using Atom

**Comment by Arcanemagus** [comment]
The error showing in your latest GIF there is a duplicate of #687. Your original image though shows `Column start (12) greater than line length (7)`, which would have to be a bug in a different rule.

**Comment by baldwmic** [comment]
@Arcanemagus ahh i shouldn't have assumed the column and line length numbers were irrelevant to the issue 😞 

So there might be two different bugs at play?

**Comment by IanVS** [comment]
Yes, it seems that `linter-eslint` has become much more sensitive to errors in ESLint rules that report incorrect column numbers.  We are fixing the ESLint rules as we find them, but first we have to figure out which rules are reporting the incorrect errors.

If you can run your original file with the eslint cli and let us know what errors you get (copy the output from eslint), that would be very helpful.

**Comment by Arcanemagus** [comment]
Yep, just like @digitalcatnip's is quite possible a different one as well 😛.

I'm not sure how, but it seems the upgrade to `atom-linter@8` from `atom-linter@5` is catching a lot more errors, even though the code checking them is essentially the same. It means the base linters (ESLint here) can be made better, but it does mean a bit of a rough time till those bugs are fixed.

**Comment by Arcanemagus** [comment]
To provide a bit more context: This linter uses the `rangeFromLineNumber` function from `atom-linter`. When it is fed a range it checks it against the file to see if it's valid and throws an error if it isn't. In the case of your original message somehow ESLint reported a message with a character of 13, on a line with only 8 characters in it. (The error message is 0-indexed, like Atom's internal coordinates)

**Comment by baldwmic** [comment]
@IanVS @Arcanemagus 

When I run 
```
./node_modules/.bin/eslint file.js
``` 
with the same `.eslintrc` file I got no output.

I tried removing the `"react/require-extension": 0,` rule and got `The react/require-extension rule is deprecated. Please use the import/extensions rule from eslint-plugin-import instead.`

**Comment by IanVS** [comment]
@baldwmic and yet when you open `file.js` in Atom, you get the errors you posted in your screenshot?  

You're not using the global eslint in your `linter-eslint` config are you, by any chance?  Anything else you can do to help find the ESLint error that's causing the problems would be very helpful.

**Comment by baldwmic** [comment]
Here's my Atom `linter-eslint` settings:

![screen shot 2016-09-09 at 15 42 09](https://cloud.githubusercontent.com/assets/10538297/18400472/0ae7ad3a-76a4-11e6-9ba8-8b90681d2caf.png)

so I think the answer is `no` I'm not using the global eslint.

**Comment by dmueller39** [comment]
I was getting errors due to this: https://github.com/eslint/eslint/pull/7088, https://github.com/eslint/eslint/issues/7086.

**Comment by lmj0011** [comment]
![line_length](https://cloud.githubusercontent.com/assets/9396189/18407408/43203da8-76d5-11e6-851c-1c619f3004b9.png)

error is still happening even after updating to eslint@3.5.0: https://github.com/AtomLinter/linter-eslint/issues/687#issuecomment-246034710

linter-eslint@7.2.4 doesn't produce this bug

**Comment by dmueller39** [comment]
Here's a PR to expose underlying errors: https://github.com/AtomLinter/linter-eslint/pull/694

**Comment by IanVS** [comment]
@lmj0011, the only thing fixed in eslint `3.5.0` is the `no-multiple-empty-lines` rule.  The one you're seeing is not from that rule.

@dmueller39, great idea, I'll review and merge ASAP.

**Comment by purplecones** [comment]
Has this been resolved? I have the linter-eslint 7.3.0 package installed and I get this error as well.

**Comment by Arcanemagus** [comment]
v7.3.1 is released which should show the offending error in the message shown to you.

As the bug causing the originally reported issue has been fixed in `eslint@3.5.0` I'm locking this issue. Please file a _new_ issue here if you need help tracking down where the problem is, or preferably over on [ESLint](https://github.com/eslint/eslint/issues) as you should now be able to tell what the offending rule is 😉.

Thanks everyone for taking the time to help track this down and report it!
